File: src/main/java/com/example/appjamback/domain/user/facade/UserFacade.java
package com.example.appjamback.domain.user.facade;

import com.example.appjamback.domain.user.domain.User;
import com.example.appjamback.domain.user.domain.repository.UserRepository;
import com.example.appjamback.domain.user.exception.UserNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.core.context.SecurityContextHolder;

@RequiredArgsConstructor
@Configuration
public class UserFacade {

    private final UserRepository userRepository;

    public User getCurrentUser() {
        String accountId = SecurityContextHolder.getContext().getAuthentication().getName();
        return getUserByAccountId(accountId);
    }

    public User getUserByAccountId(String accountId) {
        return userRepository.findByAccountId(accountId)
                .orElseThrow(() -> UserNotFoundException.EXCEPTION);
    }
}

File: src/main/kotlin/com/example/instargram/domain/user/facade/UserFacade.kt
package com.example.instargram.domain.user.facade

import com.example.instargram.domain.user.domain.User
import com.example.instargram.domain.user.domain.repository.UserRepository
import com.example.instargram.domain.user.exception.UserNotFoundException
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class UserFacade(
    private val userRepository: UserRepository
) {
    fun getCurrentUser(): User? {
        val accountId = SecurityContextHolder.getContext().authentication.name
        SecurityContextHolder.clearContext()
        return getUserByAccountId(accountId)
    }

    fun getUserByAccountId(accountId: String): User? =
        userRepository.findByAccountId(accountId)?: throw UserNotFoundException
}
